pyam/plotting.py

```diff
@@ -375,19 +375,103 @@ def stack_plot(df, x='year', y='value', stack='variable',
     # long form to one column per bar group
     _df = reshape_bar_plot(df, x, y, stack)
 
+    # Line below is for interpolation. On datetimes
+    # I think you'd downcast to seconds first and
+    # then cast back to datetime at the end..?
+    _df.index = _df.index.astype(float)
+
+    time_original = _df.index.values
+    first_zero_times = pd.DataFrame(index=["first_zero_time"])
+    both_positive_and_negative = _df.apply(
+        lambda x: (x >= 0).any() and (x < 0).any()
+    )
+
+    for col in _df.loc[:, both_positive_and_negative]:
+        values = _df[col].values
+        for i, val in enumerate(values[:-1]):
+            if np.sign(val) == -1 * np.sign(values[i + 1]):
+                x_1 = time_original[i]
+                x_2 = time_original[i + 1]
+                y_1 = val
+                y_2 = values[i + 1]
+
+                first_zero_time = x_1 - y_1 * (x_2 - x_1) / (y_2 - y_1)
+                first_zero_times.loc[:, col] = first_zero_time
+                if first_zero_time not in _df.index.values:
+                    _df.loc[first_zero_time, :] = np.nan
+
+    first_zero_times = first_zero_times.sort_values(
+        by="first_zero_time",
+        axis=1,
+    )
+    first_zero_times
+    _df = _df.reindex(sorted(_df.index)).interpolate(method="values")
+
+    # Sort lines so that negative timeseries are on the right,
+    # positive timeseries are on the left and timeseries which
+    # go from positive to negative are ordered such that the
+    # timeseries which goes negative first is on the right
+    # (case of timeseries which go from negative to positive
+    # is an edge case we haven't thought about as it's unlikely
+    # to apply to us).
+    col_order = [
```

Comment:
I think this can be cleaned up a bit too, e.g.,

```
col_order = [c for c in _df if (_df[c] > 0).all()] + first_zero_times.columns[::-1] + [c for c in _df if (_df[col] < 0).all()] 
```